src/frontend/next/src/components/SearchPage/SearchResults.tsx

```diff
@@ -0,0 +1,85 @@
+import { FC } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { useSWRInfinite } from 'swr';
+import { Container, Box, Grid, Typography, ListSubheader } from '@material-ui/core';
+
+import useSiteMetadata from '../../hooks/use-site-metadata';
+import Timeline from '../Posts/Timeline.jsx';
+import Spinner from '../Spinner/Spinner';
+
+interface SearchPageProps {
+  text: string | null;
+  filter: string | null;
+}
+
+const useStyles = makeStyles(() => ({
+  spinner: {
+    display: 'flex',
+    justifyContent: 'center',
+  },
+  searchResults: {
+    padding: 0,
+    width: '100%',
+    justifyContent: 'center',
+  },
+}));
+
+const SearchResults: FC<SearchPageProps> = ({ text, filter }: SearchPageProps) => {
+  const classes = useStyles();
+  const { telescopeUrl } = useSiteMetadata();
+  const prepareUrl = (index: number) =>
+    text
+      ? `${telescopeUrl}/query?text=${encodeURIComponent(text)}&filter=${filter}&page=${index}`
+      : // Should we return an empty string? or null here?
+        null;
+  // We only bother doing the request if we have something to search for.
+  // Can't do optional chaining below on text only because text could be null
+  const shouldFetch = () => text && text.length > 0;
+  const { data, size, setSize, error } = useSWRInfinite(
+    (index: number) => (shouldFetch() ? prepareUrl(index) : null),
+    async (u: string) => {
+      const res = await fetch(u);
+      const results = await res.json();
+      return results.values;
+    }
+  );
+
+  if (error) {
+    return (
+      <Container className={classes.searchResults}>
+        <Box className={classes.root} boxShadow={2} marginTop={10}>
```

Comment:
Getting `Property 'root' does not exist on type 'Record<"spinner" | "searchResults", string>'.` with the latest master merge, same with line 52 and 53.

But I'm getting `No search results` when rendering the component so I think it's fine.